.github/workflows/windows-ci.yml

```diff
@@ -0,0 +1,89 @@
+---
+name: Windows build and test
+on:
+  push:
+    branches:
+      - 'master'
+  pull_request:
+    branches:
+      - '*'
+  release:
+
+jobs:
+  test:
+    name: Run tests
+    runs-on: windows-2019
+    strategy:
+      fail-fast: false
+      matrix:
+        include: [
+          { arch: x86_64, msystem: mingw64, tests: none },
+          { arch: x86_64, msystem: mingw64, tests: main },
+          { arch: i686, msystem: mingw32, tests: main },
+          { arch: i686, msystem: mingw32, tests: none },
+        ]
+    steps:
+      - name: Checkout source
+        uses: actions/checkout@v2
+        with:
+          path: ${{ github.event.repository.name }}
+      - name: Fetch Data-Sample repository
+        uses: actions/checkout@v2
+        with:
+          repository: coin-or-tools/Data-Sample
+          path: Data/Sample
+      - name: Fetch ThirdParty-Glpk repository
+        uses: actions/checkout@v2
+        with:
+          repository: coin-or-tools/ThirdParty-Glpk
+          path: ThirdParty/Glpk
+      - name: Checkout coinbrew
+        uses: actions/checkout@v2
+        with:
+          repository: coin-or/coinbrew
+          path: coinbrew
+      - name: Set correct host flag and install requirements
+        run: |
+          echo "host_flag=--host=${{ matrix.arch }}-w64-mingw32" | Out-File -FilePath $env:GITHUB_ENV -Encoding utf8 -Append
+          C:\msys64\usr\bin\pacman -S mingw-w64-${{ matrix.arch }}-lapack mingw-w64-${{ matrix.arch }}-winpthreads-git mingw-w64-${{ matrix.arch }}-readline mingw-w64-${{ matrix.arch }}-suitesparse mingw-w64-${{ matrix.arch }}-metis --noconfirm
+      - name: Set up msys with ${{ matrix.msystem }}
+        uses: msys2/setup-msys2@v2
+        with:
+          update: true
+          install: >-
+            base-devel
+            git
+          path-type: inherit
+          msystem: ${{ matrix.msystem }}
+      - name: Build project
+        run: |
+          [[ ${{ matrix.tests }} == "none" ]] && debug="--enable-debug" suffix="-dbg" || debug="" suffix=""
```

Comment:
Doesn't this say "build debug if no tests"? Did you want the other way around?

Reply:
I literally just copied what appveyor was doing, which is this:
```
```

Reply:
If you think the other way makes more sense, I'll change it now.

Reply:
na fine, maybe there was a reason

Reply:
If it's not obvious to us right now, then that reason doesn't matter. I think your suggestion is a good one, and I'm going to implement it unless we figure out a compelling reason not to.

Reply:
@tkralphs you implemented this originally: https://github.com/coin-or/CoinUtils/commit/1a0231654bc5a1d45630eb0270b6cfb803beb033

Is there a reason you did it this way?

Reply:
I don't see the connection between what was done in Appveyor and the logic Stefan is questioning. What you quoted from Appveyor doesn't contain any matrix logic. It's just defining one build configuration by explicitly enumerating the settings for that one  configuration, which is "build binaries with debugging enabled and don't do any tests." If we turn that into a matrix instead of explicitly enumerating, then Stefan's logic is essentially what I would do to replicae the same thing: "when debugging is enabled, don't do any tests." 

But with that said, skipping testing was probably just to save time on the Appveyor builds, although there may have been some other reason that we'll find out when we try testing the debug builds. One would generally only grab the debug binaries if the regular ones were failing in some way, so one could argue that testing debug binaries is not that important. On the other hand, if build time is no longer an issue, there is some advantage to testing debug binaries, since they have asserts that the regular binaries do not.

Bottom, line, let's try testing debug builds, too, and see what happens.

Reply:
I'm not sure I understand your concern about matrix logic. From my perspective, the question we are trying to answer is "should we preserve the same build configuration (where we did not test debug binaries)" or "should we move to a different build configuration {where debug binaries always run tests)". I think we agree that the second thing is what we want, so I'm going to push that commit now.

Reply:
The new version is now doing debug build iff testing. And you do two builds with testing and not-testing.
We can of course leave it like that, but maybe it would be easier to read when just decoupling test/no-test and debug/no-debug. Further, for CoinUtils, tests don't take long, so should be ok to just always run them.

Thus, would it be ok to change the test flag in the matrix to a debug flag and always run tests?

For projects further up, like Clp and especially Cbc, I can imagine that doing a test on a debug-build can take too much time.

Reply:
> I'm not sure I understand your concern about matrix logic. From my perspective, the question we are trying to answer is "should we preserve the same build configuration (where we did not test debug binaries)" or "should we move to a different build configuration {where debug binaries always run tests)". I think we agree that the second thing is what we want, so I'm going to push that commit now.

I'm not concerned about matrix logic, it's perfectly fine, I just didn't have it before and I was trying to compare the previous logic and the new logic. You said you just directly translated what was in Appveyor, but the Appveyor didn't actually have this logic because it wasn't matrixed. I was only trying to point this out to explain. 

The build configurations, if matrixed, should be effectively "debug" and "release". Then there should be logic saying what happens with a debug build and a release build as a secondary thing that could change over time. The way it's currently matrixed, we instead have "test" and "don't test" at the highest level and then there is logic to say that if we don't test, then do a debug build. So it's backward as Stefan pointed out. If want to test debug builds at some point, we have to change the whole matrix, whereas that should only be a change to the procedure for doing debug builds. 

Does it make more sense now?

Reply:
Sorry, I don't know what you want. Please feel free to push to this branch so that your preferred configuration is run.

Reply:
Or, if this is something that requires further discussion, that discussion should happen in an issue and maybe not in this PR. I wanted to keep the config as faithful as possible to the original appveyor config, while this perhaps isn't the most optimal iteration, is it better than what it is replacing?